### ISSUE TITLE
Parallel vs Queue incoherent behaviour

### ISSUE THREAD

**Issue by FMaz008** [Chia-Network/chia-blockchain-gui#424]
I find it confusing and it makrs it hard to have parallel queues.

Here is my more recent experience om V1.1.16 with win10:

**I started 2 parallel plots. Then I added a plot of 10 queues. Unexpectedly, I ended having 3 plots plotting.** 

I restarted, deleted the plots and started over. 

_When I openned the GUI I got the "Connecting to Wallet" for over 30min. Realize it was an infinite loop of some kind. Had to google it and delete wallet/blockchain_* files manually. Then the GUI launched._

So I am thinking the solution would be to **add 6 plots to queue, twice. I did that but this time when I added 6 plots to the queue the second time they just added to the queues, so I ended up with only 1 plot plotting and 11 queued.**

I tried to see if there was a way to click on a queued item and tell it to "start now" there isn't.

So from there I was confused. I ended up adding a single parallel plot. Not sure once it finish if it will pick something from the queue of be its own queue.

One suggestion would be to have the queues represented better visually (ie: a tab per queue?) And have a setting "number of parallel plots" for each tab.

That way if I have 1 Sata SSD and 1 NVME SSD I can have a queue for each with different parallel plots.

Hope this makes sense!
Francois

**Comment by padik11** [comment]
I confirm. We need to implement a queue for the queue. When the first plot in the first queue is saved, the second queue starts.

**Comment by FMaz008** [comment]
Just to clarify: I just come to realize the hidden advance section has the queue name setting. So the behavior I am describing was happening with the default name.

**Comment by padik11** [comment]
Does this mean that if I give the same name on the second round, she'll be waiting for the first round to finish?

**Comment by FMaz008** [comment]
> Does this mean that if I give the same name on the second round, she'll be waiting for the first round to finish?

Well, my concern was just that there a missmatch in behaviour where as if you do the same action (adding a new plot to a queue), you will get a different result depending on how you add it.

As I understand it, the intended behavior is that if you give the same queue name, it will append what you add to the queue, unless you ad parallel item, in which case it will start them up immediately.

It could be just the inteface that is not clear enough (ie: move the queue name setting out of the advanced section), or a bug.

The annoying part is that you can't delete queued item or order queued item to start.

**Comment by silence48** [comment]
the parallel queue and the queue queue are in effect 2 different queues.  because of this, if you start a queue called 'default' as parallel and set it to do 2 at a time, then create a new queue called 'default' it will not add those plots to the end of the parallel queue... I suggest starting 2 separate queues with different names under the advanced options.

**Comment by FMaz008** [comment]
> the parallel queue and the queue queue are in effect 2 different queues. because of this, if you start a queue called 'default' as parallel and set it to do 2 at a time, then create a new queue called 'default' it will not add those plots to the end of the parallel queue, but rather have the 2 parallel running then also run the first one from the new queue... I suggest starting 2 separate queues with different names under the advanced options, and not using the parallel option. As long as they have separate names they will still run in parallel, and you should also stagger the start.

Make complete sense.
I think the UI confused me.
Maybe moving the Queue name from the advanced section to the section where we can choose the queue type would improve the user friendliness?

**Comment by silence48** [comment]
> > the parallel queue and the queue queue are in effect 2 different queues. because of this, if you start a queue called 'default' as parallel and set it to do 2 at a time, then create a new queue called 'default' it will not add those plots to the end of the parallel queue, but rather have the 2 parallel running then also run the first one from the new queue... I suggest starting 2 separate queues with different names under the advanced options, and not using the parallel option. As long as they have separate names they will still run in parallel, and you should also stagger the start.
> 
> Make complete sense.
> I think the UI confused me.
> Maybe moving the Queue name from the advanced section to the section where we can choose the queue type would improve the user friendliness?

I completely agree there should be some type of change.  It would be nice to have an option to make a 'queue of queues' of sorts, that could allow us to automatically stagger the start of plots, doing automatically what gui users must do manually.  There are of course third party utilities to accomplish this such as "swar" or "plotman"
